RockWeb/Blocks/Core/MergeTemplateEntry.ascx.cs

```diff
@@ -140,6 +147,7 @@ protected void btnMerge_Click( object sender, EventArgs e )
             // NOTE: This is a full postback (not a partial like most other blocks)
 
             var rockContext = new RockContext();
+            rockContext.Database.CommandTimeout = 600;
```

Comment:
@tcavaletto can you step through an example of the problem you're seeing and try to isolate where/when the timeout occurs?  If it's only in the Merge_Click of the MergeTemplateEntry block then, a block setting with a default of 600 is probably a good way to go.